### PR TITLE
feat: update aws-cli to an OIDC role compatible version

### PR DESCRIPTION
**Pull request by luphaz**
This PR aims to source default credentials from OIDC role in drone Kubernetes pipelines.

This will not change the final role used (assume_role property) but source initial credentials from OIDC rather than EC2 instance profile.

The EC2 instance profile will still be used in NON Kubernetes drone pipelines.

This PR will also update drone pipelines to Kubernetes as we are starting the migration.

This PR has been tested here by created a docker and Kubernetes drone pipelines to build the Kalyke control plane AMIs : 
* https://drone.heetch.net/heetch/cosmos/10412/1/2
* https://drone.heetch.net/heetch/cosmos/10421/2/1

----

AWS-cli will automatically the appropriate credentials depending on env var definition and role given if the version is high enough to pick it : 
* https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-minimum-sdk.html
* AWS CLI — 1.16.232

 In order to enable OIDC role sourcing, update aws-cli version to latest aws-cli v1 version available

Current version is : 
```text
aws-cli/1.14.5 Python/2.7.13 Linux/4.19.76-linuxkit botocore/1.8.9
```

New version will be : 
```text
aws-cli/1.18.165 Python/3.8.5 Linux/4.19.76-linuxkit botocore/1.19.5
```

> NB: we are not migrating to aws-cli v2 as it's built against `glibc` that is not contained into alpine by default
> here is the smallest change we can do to keep update a retro-compatible as possible
> here is an [example](https://stackoverflow.com/questions/60298619/awscli-version-2-on-alpine-linux) on how to install aws-cli v2 on alpine

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
     openssl \
     ca-certificates \
     openssh
 
+RUN apk add --no-cache \
+    python3 \
+    py3-pip \
+    && pip3 install --upgrade pip \
+    && pip3 install \
+    awscli \
```
Comment (WinterNis) [inline Dockerfile:13]: :tulip: Could we pin the version ?

Comment (luphaz) [reply Dockerfile:13]: Indeed let me check 👍

Comment (gyndav) [reply Dockerfile:13]: no requirements.txt for deps management?

Comment (luphaz) [reply Dockerfile:13]: We could indeed even if we have a single lib for now
Will change the pin version to use requirements.txt

**Review by WinterNis — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (luphaz) [reply Dockerfile]: I pined the version into `requirements.txt` let me know if it's what you had in mind.

**Review by WinterNis — APPROVED** [review]

**Review by gyndav — APPROVED** [review]

**Pull request merged**